Shootingat Resolved Project/Assets/Scripts/Characters/Core/Stats/Stat.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using TMPro;
+using UnityEngine;
+
+namespace PabloLario.Characters.Core.Stats
+{
+
+    // Forcing generic type to be number is not available until C# 10
+    // Remove FlaotStat and intStat then
+    //https://stackoverflow.com/a/68736791/14559140
+    [System.Serializable]
+    public abstract class Stat<T>
+    {
+
+
+        public abstract T Value { get; set; }
+
+        [SerializeField]
+        protected T limitValue;
+
+        public T LimitValue { get => limitValue; }
+
+        [SerializeField]
+        protected bool improvingIncreasesValue = true;
+
+        [SerializeField]
+        private TMP_Text textUi;
+
+        [SerializeField]
+        private string textFixture;
+
+        public delegate void OnUpdate(Stat<T> previous, Stat<T> after);
+        public OnUpdate onUpdateValue;
+
+
+        protected void UpdateText()
```

Comment:
Need to refactor this. It is responsible of both UI and stats right now. That's not good